src/tools/compiletest/src/header.rs

```diff
@@ -704,12 +736,31 @@ impl Config {
                 Some(Debugger::Gdb) => name == "gdb",
                 Some(Debugger::Lldb) => name == "lldb",
                 None => false,
-            };
+            } || name == "cfg" && self.cfg_matches_for_target(maybe_value.unwrap(), target);
+
+        if is_match { EvaluatedProp::Match } else { EvaluatedProp::NoMatch }
+    }
+
+    fn cfg_matches_for_target(&self, value: &str, target: &str) -> bool {
+        let whitespace_or_double_quote = |c: char| c.is_whitespace() || c == '"';
+
+        let value = value.replace(whitespace_or_double_quote, "");
 
-        if is_match { ParsedNameDirective::Match } else { ParsedNameDirective::NoMatch }
+        let cfg_data = Command::new(&self.rustc_path)
+            .args(&["--target", &target])
+            .args(&["--print", "cfg"])
+            .output()
+            .unwrap()
+            .stdout;
+        let cfg_data = String::from_utf8(cfg_data).unwrap();
+        let cfg_data = cfg_data.replace('"', "");
```

Comment:
The `cfg_data` could presumably be cached somewhere around this point. Doing so could presumably result in some small performance saving for the case where `target == self.target`, which is likely to come up with the most frequency. I'm not sure whether this is worthwhile, or if so, how to go about it.

As an amusing anecdote, I actually fiddled around with a version where all of the targets had their information calculated in advance and it was stored in `Config`. I figured that computing all of that would only take a few extra seconds. It took me a few days to figure out why doing so was causing a one third to one half regression in testing times. It turns out that `Config` gets cloned a fair number of times, and by the end was taking enough memory that it was vastly increasing the duration of forking the threads to execute the tests.